packages/ra-supabase-core/src/dataProvider.ts

```diff
@@ -36,7 +45,50 @@ export const supabaseDataProvider = ({
         primaryKeys,
         schema,
     };
-    return postgrestRestProvider(config);
+    return withLifecycleCallbacks(postgrestRestProvider(config), [
```

Comment:
**polish**: We should provide some customization to users who use this library. This implementation does not allow this.

See [this PR discussion](https://github.com/marmelab/ra-supabase/pull/52#issuecomment-1959465400) for more information.